osu.Game.Rulesets.Mania/Beatmaps/Patterns/Legacy/HitObjectPatternGenerator.cs

```diff
@@ -303,8 +303,7 @@ private Pattern generateRandomPatternWithMirrored(double centreProbability, doub
 
             var pattern = new Pattern();
 
-            bool addToCentre;
-            int noteCount = getRandomNoteCountMirrored(centreProbability, p2, p3, out addToCentre);
+            int noteCount = getRandomNoteCountMirrored(centreProbability, p2, p3, out var addToCentre);
```

Comment:
Maybe `out bool` would be better here, since this is the only instance wherein the type is not immediately obvious from the function call/variable name?

Reply:
This could be handled by "don't use var for predefined type". I just applied the auto fix here.

Reply:
I think the guideline for `var` was that it should be used only if the type can be inferred from the context? Which is unfortunately kind of hard to enforce with an analyzer, true. I'd rather not enable "don't use var for predefined type", since the same commit here shows examples where that doesn't make a lot of sense (those `int.Parse` calls have no reason to then have `out int`)

Reply:
The code base doesn't have wide consistency about `var` now. I'd prefer it discussed separately.